gradoop-flink/src/main/java/org/gradoop/flink/algorithms/gelly/hits/HITS.java

```diff
@@ -0,0 +1,118 @@
+ * convergence threshold, or both.
+ * <p>
+ * The Results are stored as properties of the vertices (with given keys).
+ */
+public class HITS extends GellyAlgorithm<NullValue, NullValue> {
+
+  /**
+   * Property key to store the authority score.
+   */
+  private String authorityPropertyKey;
+
+  /**
+   * Property key to store the hub score.
+   */
+  private String hubPropertyKey;
+
+  /**
+   * Gelly HITS implementation
+   */
+  private org.apache.flink.graph.library.link_analysis.HITS<GradoopId, NullValue, NullValue> hits;
+
+  /**
+   * HITS with fixed number of iterations
+   *
+   * @param authorityPropertyKey Property key to store the authority score.
+   * @param hubPropertyKey       Property key to store the hub score.
+   * @param iterations           number of iterations
+   */
+  public HITS(String authorityPropertyKey, String hubPropertyKey, int iterations) {
+    super(new VertexToGellyVertexWithNullValue(), new EdgeToGellyEdgeWithNullValue());
+    this.authorityPropertyKey = authorityPropertyKey;
+    this.hubPropertyKey = hubPropertyKey;
+    hits = new org.apache.flink.graph.library.link_analysis.HITS<>(iterations, Double.MAX_VALUE);
+  }
+
+
+  /**
+   * HITS with convergence threshold
+   *
+   * @param authorityPropertyKey Property key to store the authority score.
+   * @param hubPropertyKey       Property key to store the hub score.
+   * @param convergenceThreshold convergence threshold for sum of scores
+   */
+  public HITS(String authorityPropertyKey, String hubPropertyKey, double convergenceThreshold) {
+    super(new VertexToGellyVertexWithNullValue(), new EdgeToGellyEdgeWithNullValue());
+    this.authorityPropertyKey = authorityPropertyKey;
+    this.hubPropertyKey = hubPropertyKey;
+    hits = new org.apache.flink.graph.library.link_analysis.HITS<>(Integer.MAX_VALUE,
+      convergenceThreshold);
+  }
+
+  /**
+   * HITS with convergence threshold and maximum number of iterations
+   *
+   * @param authorityPropertyKey Property key to store the authority score.
+   * @param hubPropertyKey       Property key to store the hub score.
+   * @param convergenceThreshold convergence threshold for sum of scores
+   * @param maxIterations        maximum number of iterations
+   */
+  public HITS(String authorityPropertyKey, String hubPropertyKey, int maxIterations,
+    double convergenceThreshold) {
+    super(new VertexToGellyVertexWithNullValue(), new EdgeToGellyEdgeWithNullValue());
+    this.authorityPropertyKey = authorityPropertyKey;
+    this.hubPropertyKey = hubPropertyKey;
+    hits =
+      new org.apache.flink.graph.library.link_analysis.HITS<>(maxIterations, convergenceThreshold);
+  }
+
+
+  @Override
+  protected LogicalGraph executeInGelly(Graph<GradoopId, NullValue, NullValue> graph) throws
+    Exception {
+
+    DataSet<Vertex> newVertices =
+      hits.run(graph).join(currentGraph.getVertices()).where(0).equalTo(new Id<>())
```

Comment:
we use a specific call style that can sadly not be enforced with checkstyle. It's:
```
firstDataFrame
.transformation(otherDataFrame)
.call
.call
```
which leads to
```
DataSet<Vertex> newVertices = hits.run
  .join(currentGraph.getVertices)
  .where(0).equalTo(new Id<>())
  .with(new HITSToAttributes(...)
```